File: model/src/main/scala/ru/mipt/acsl/decode/model/proxy/DecodeProxyResolver.scala
package ru.mipt.acsl.decode.model.proxy

import ru.mipt.acsl.decode.model.Referenceable
import ru.mipt.acsl.decode.model.proxy.path.ProxyPath
import ru.mipt.acsl.decode.model.registry.Registry

/**
  * @author Artem Shein
  */
trait DecodeProxyResolver {
  def resolveElement(registry: Registry, path: ProxyPath): (Option[Referenceable], ResolvingResult)
}

File: model/src/main/java/ru/mipt/acsl/decode/model/proxy/DecodeProxyResolver.java
package ru.mipt.acsl.decode.model.proxy;

import ru.mipt.acsl.decode.model.Referenceable;
import ru.mipt.acsl.decode.model.proxy.path.ProxyPath;
import ru.mipt.acsl.decode.model.registry.Registry;

/**
 * @author Artem Shein
 */
public interface DecodeProxyResolver {

    ResolvingResult<Referenceable> resolveElement(Registry registry, ProxyPath path);

}
